superdesk/vocabularies/async_vocabularies.py

```diff
@@ -0,0 +1,486 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2025 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+
+import json
+import logging
+from typing import Annotated, Any, Dict, List, Optional
+from typing import List
+
+from eve.methods.common import serialize_value
+from pydantic import BaseModel, ConfigDict, Field
+from quart_babel import gettext as _, lazy_gettext
+
+from superdesk import privilege
+from superdesk.cache import cache
+from superdesk.core.resources import ResourceModel, dataclass
+from superdesk.core.resources.model import ResourceModel
+from superdesk.core.resources.service import AsyncResourceService
+from superdesk.core.resources.validators import validate_maxlength
+from superdesk.core.types import SearchRequest
+from superdesk.default_schema import DEFAULT_EDITOR, DEFAULT_SCHEMA
+from superdesk.errors import SuperdeskApiError
+from superdesk.flask import request
+from superdesk.notification import push_notification
+from superdesk.resource_fields import ID_FIELD
+from superdesk.users import get_user_from_request
+from superdesk.utc import utcnow
+
+logger = logging.getLogger(__name__)
+
+KEYWORDS_CV = "keywords"
+
+
+privilege(
+    name="vocabularies",
+    label=lazy_gettext("Vocabularies Management"),
+    description=lazy_gettext("User can manage vocabularies' contents."),
+)
+
+
+# TODO(petr): add api to specify vocabulary schema
+vocab_schema = {
+    "crop_sizes": {
+        "width": {"type": "integer"},
+        "height": {"type": "integer"},
+    }
+}
+
+
+@dataclass
+class Tag:
+    text: str
+
+
+class Item(BaseModel):
+    model_config = ConfigDict(extra="allow")
+    name: str
+    qcode: str
+    is_active: bool = True
+
+
+@dataclass
+class DateShortcut:
+    value: int
+    term: str
+    label: str
+
+
+@dataclass
+class CustomFieldConfig:
+    increment_steps: list[int]
+    initial_offset_minutes: int
+
+
+class VocabulariesResourceModel(ResourceModel):
+    display_name: str
+    description: str | None = None
+    helper_text: Annotated[str | None, validate_maxlength(120)] = None
+    tags: list[Tag] | None = None
+    popup_width: int | None = None
+    type_: str  # Use type_ instead of type due to reserved keyword conflict
```

Comment:
We need to make this an alias, so the DB and front-end still stores the item in the same dict key. We can then use a name that provides context too. Something like 
```python
from enum import Enum, unique

@unique
class CVAccessType(str, Enum):
    MANAGEABLE ="manageable"
    UNMANAGEABLE = "unmanageable"

class VocabulariesResourceModel(ResourceModel):
    ...
    management_type: Annotated[CVAccessType, Field(alias="type")]
```